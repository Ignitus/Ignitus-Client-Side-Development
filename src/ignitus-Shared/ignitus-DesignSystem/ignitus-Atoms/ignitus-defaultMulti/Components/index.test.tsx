import React from 'react';
import { mount, render } from 'enzyme';

import { DefaultMultiMediaInput } from './index';

const props = {
  placeholder: 'Placeholder',
  name: 'Name',
  maxHeightOfInput: 200,
};

it('should render', () => {
  const tree = render(<DefaultMultiMediaInput {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DefaultMultiMediaInput />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<DefaultMultiMediaInput {...props} />);
  });

  it('should contain buttons & textarea', () => {
    // one button with text Post
    expect(wrapper.findWhere(n => n.text() === 'Post').length).toBe(1);

    // other 5 buttons with svg
    expect(wrapper.find('button svg').length).toBe(5);
    expect(wrapper.find('textarea').length).toBe(1);
  });
});
