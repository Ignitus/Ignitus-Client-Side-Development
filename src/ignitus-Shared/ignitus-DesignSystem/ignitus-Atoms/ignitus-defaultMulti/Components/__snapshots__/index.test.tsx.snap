// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  class="css-fch366"
>
  <div
    class="css-k2464f"
  >
    <textarea
      class="css-k7ncnn"
      name="Name"
      placeholder="Placeholder"
    />
  </div>
  <div
    class="css-bj1psw"
  >
    <div
      class="css-1oeiakk"
    >
      <button
        class="css-qkkh99"
      >
        <svg
          aria-label="Library Books Icon"
          class="css-1cood3v"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Library Books Icon
          </title>
          <path
            d="M3 6c-.55 0-1 .45-1 1v13c0 1.1.9 2 2 2h13c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1-.45-1-1V7c0-.55-.45-1-1-1zm17-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 9h-8c-.55 0-1-.45-1-1s.45-1 1-1h8c.55 0 1 .45 1 1s-.45 1-1 1zm-4 4h-4c-.55 0-1-.45-1-1s.45-1 1-1h4c.55 0 1 .45 1 1s-.45 1-1 1zm4-8h-8c-.55 0-1-.45-1-1s.45-1 1-1h8c.55 0 1 .45 1 1s-.45 1-1 1z"
          />
        </svg>
      </button>
      <button
        class="css-qkkh99"
      >
        <svg
          aria-label="Poll Icon"
          class="css-1cood3v"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Poll Icon
          </title>
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM8 17c-.55 0-1-.45-1-1v-5c0-.55.45-1 1-1s1 .45 1 1v5c0 .55-.45 1-1 1zm4 0c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8c0 .55-.45 1-1 1zm4 0c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1z"
          />
        </svg>
      </button>
      <button
        class="css-qkkh99"
      >
        <svg
          aria-label="Pdf Icon"
          class="css-1cood3v"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Pdf Icon
          </title>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8.5 7.5c0 .83-.67 1.5-1.5 1.5H9v2H7.5V7H10c.83 0 1.5.67 1.5 1.5v1zm5 2c0 .83-.67 1.5-1.5 1.5h-2.5V7H15c.83 0 1.5.67 1.5 1.5v3zm4-3H19v1h1.5V11H19v2h-1.5V7h3v1.5zM9 9.5h1v-1H9v1zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm10 5.5h1v-3h-1v3z"
          />
        </svg>
      </button>
      <button
        class="css-qkkh99"
      >
        <svg
          aria-label="Video Library Icon"
          class="css-1cood3v"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Video Library Icon
          </title>
          <path
            d="M3 6c-.55 0-1 .45-1 1v13c0 1.1.9 2 2 2h13c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1-.45-1-1V7c0-.55-.45-1-1-1zm17-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8 12.5v-9l5.47 4.1c.27.2.27.6 0 .8L12 14.5z"
          />
        </svg>
      </button>
      <button
        class="css-qkkh99"
      >
        <svg
          aria-label="Camera Icon"
          class="css-1cood3v"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Camera Icon
          </title>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 15.2C13.7673 15.2 15.2 13.7673 15.2 12C15.2 10.2327 13.7673 8.8 12 8.8C10.2327 8.8 8.8 10.2327 8.8 12C8.8 13.7673 10.2327 15.2 12 15.2ZM9 2L7.17 4H4C2.9 4 2 4.9 2 6V18C2 19.1 2.9 20 4 20H20C21.1 20 22 19.1 22 18V6C22 4.9 21.1 4 20 4H16.83L15 2H9ZM12 17C9.24 17 7 14.76 7 12C7 9.24 9.24 7 12 7C14.76 7 17 9.24 17 12C17 14.76 14.76 17 12 17Z"
          />
        </svg>
      </button>
    </div>
    <button
      class="css-183d2ex"
    >
       Post
    </button>
  </div>
</div>
`;
